src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -364,6 +386,32 @@ - (void)issueClosedWithTrailingMetadata:(NSDictionary *)trailingMetadata error:(
   }
 }
 
+- (void)issueDidWriteData {
+  @synchronized(self) {
+    if (_callOptions.enableFlowControl && [_handler respondsToSelector:@selector(didWriteData)]) {
+      dispatch_async(_dispatchQueue, ^{
+        id<GRPCResponseHandler> copiedHandler = nil;
+        @synchronized(self) {
+          copiedHandler = self->_handler;
+        };
+        [copiedHandler didWriteData];
+      });
+    }
+  }
+}
+
+- (void)receiveNextMessages:(NSUInteger)numberOfMessages {
+  GRPCCall *copiedCall = nil;
```

Comment:
```
if (!_callOptions.enableFlowControl) {
  return;
}
```
This function is a noop if flow control is disabled on the call. Similar comment for `initWithRequestOption` above.

Reply:
no-op behavior is controlled inside `GRPCCall`, so I think it doesn't really matter at this place. But for readability I'll add it here.

Reply:
Yes, it don't affect correctness, just an optimization as we avoid an unnecessary function call.

Reply:
Actually I think I'll add comment here to explain

Reply:
No there's a difference. The options for the call converge in `GRPCCall` object. It is possible to call the `receiveNextMessages:` function before a call starts, at which time the `_callOptions` object is still not initialized. I think it is better to let `GRPCCall` handle it.

Reply:
`_callOptions` is initialized in GRPCall2's constructor (`initWithRequestOptions`). When the `GRPCCall` object is created in `start`, its `_callOptions` is initialized with `GRPCCall2`'s callOptions.
But anyway, your code is fine as is, a comment to explain should be good enough.